src/Microsoft.ML.Data/DataView/RowToRowMapperTransform.cs

```diff
@@ -217,6 +217,11 @@ public override DataViewRowCursor[] GetRowCursorSet(IEnumerable<DataViewSchema.C
         void ISaveAsOnnx.SaveAsOnnx(OnnxContext ctx)
         {
             Host.CheckValue(ctx, nameof(ctx));
+
+            const int minimumOpSetVersion = 9;
+            if (ctx.GetOpSetVersion() < minimumOpSetVersion)
+                throw Contracts.ExceptParam(nameof(minimumOpSetVersion), $"Requested OpSet version {ctx.GetOpSetVersion()} is lower than {LoaderSignature}'s minimum OpSet version requirement: {minimumOpSetVersion}");
+
```

Comment:
Is it necessary or meaningful to have this check here in RowToRowMapperTransform? It looks to me this transform doesn't actually require specifically any opset, as it simply calls the Mapper's SaveAsOnnx method. Because of this, it's better to leave it to the mappers to say if they support or not any given opset.

Reply:
Yeah if no Opset is used, it's not necessary to require a version check there. btw, is there any quick way to check whether it uses Opset or not?

Reply:
As I mentioned in another comment, I think that checking for opset>=9 in every component is redundant, but in case you decide to leave these checks, there are a few other components where their implementation of `SaveAsOnnx` calls a different component's implementation of `SaveAsOnnx`, similar to `RowToRowMapperTransform`. These implementations shouldn't do the checks either. Here are the components I found that call other components' implementation of `SaveAsOnnx` (please verify if there are others that I missed):

`CalibratorTransformer` (in CalibratorCatalog.cs)
`GenericScorer`
`MulticlassClassificationScorer`
`PredictedLabelScorer`
`SchemaBindablePredictorWrapper`
`SequentialTransformerBase`




---
In reply to: [432971026](https://github.com/dotnet/machinelearning/pull/5176#discussion_r432971026) [](ancestors = 432971026)